internal/commands/crypto.go

```diff
@@ -246,17 +250,45 @@ func newCryptoGenerateCmd(ctx *CmdCtx, category, algorithm string) (cmd *cobra.C
 // CryptoRandRunE is the RunE for the authelia crypto rand command.
 func (ctx *CmdCtx) CryptoRandRunE(cmd *cobra.Command, args []string) (err error) {
 	var (
-		random string
+		random     string
+		quiet      bool
+		outputPath string
 	)
 
 	if random, err = flagsGetRandomCharacters(cmd.Flags(), cmdFlagNameLength, cmdFlagNameCharSet, cmdFlagNameCharacters); err != nil {
 		return err
 	}
 
-	fmt.Printf("Random Value: %s\n", random)
+	quiet, _ = cmd.Flags().GetBool(cmdFlagNameQuiet)
 
-	if value := url.QueryEscape(random); random != value {
-		fmt.Printf("Random Value (URL Encoded): %s\n", value)
+	if outputPath, err = flagsGetFilePath(cmd.Flags(), cmdFlagNameOutputFile, cmdFlagNameDirectory); err != nil {
+		return fmt.Errorf("error getting output file path: %v", err)
+	}
```

Comment:
Instead of writing a custom utility I would use the same snippet as here to get and check the file path:
https://github.com/authelia/authelia/blob/4dcf8936c22074ecb09b721a3306967121dee10a/internal/commands/storage_run.go#L439-L452

Or maybe even extract this snippet into a utility function and replace all other appearances of this snippet in the authelia CLI codebase with this new utility. But maybe that is outside the scope of this PR.